ion_phys/common.py

```diff
@@ -140,8 +140,7 @@ def __init__(self, B=None, *, I=0, levels={}, transitions={},
 
         self._sort_levels()  # arrange levels in energy order
 
-        if B is not None:
-            self.setB(B)
+        self.setB(B)
```

Comment:
`Ion` still defaults to `B=None`. I think it's sensible to handel the case `B=None` instead of raising an exception.
```suggestion        
        if B is not None:
            self.setB(B)
```

Reply:
@pathfinder49 it’s easier to track if you open issues /prs rather than comment on historic commits. 

Feel free to submit commits on the broken error messages (should not mention order, maybe just the chan in quantum numbers?).

For default B (more re @rhanley1 comment than yours) I think this needs discussion (eg a new issue). There are two issues: handle the case of 0 field where states are degenerate. I’m not sure how we should handle this or if it’s even a useful thing to do — what’s the use case for 0field? I’m not aware of any experiments that operate at strictly 0 field. 

Beyond that there is the question of default parameters. I think the principle of least surprise is important here - I’d rather have a bit more verbosity and less chance of surprising mistakes. Since I think strictly 0 field is probably a niche use case I don’t want to default to it.

Reply:
@pathfinder49 a good default behaviour would be to not calculate anything (eigenvectors matrix elements etc) if B is None. Feel free to submit a pr

Reply:
@pathfinder49 a good default behaviour would be to not calculate anything (eigenvectors matrix elements etc) if B is None. Feel free to submit a pr

Reply:
@pathfinder49 sorry, was reviewing on my phone earlier on and got confused (confused myself into thinking that this was for one of my old commits, not for this PR).

I think we're basically in agreement.

To clarify:
1. I think the default B bits need to go into a separate PR (@RHanley1 it's best to create a new branch for each PR so you can push specific commits to that PR only, rather than creating a PR from your master and merging everything into one).
2. Currently, I think that a default B would do more harm than good. It's very little extra work to say `ion = Ca43(B=1e-9)` than `Ca43()` and it makes it explicit that the simulation is low field.
3. It might be worth putting a check for B exactly equal to 0 since this currently raises an exception that's not very clear.